tests/serac_quasistatic_solver.cpp

```diff
@@ -38,64 +38,76 @@ TEST(quasistatic_solver, qs_solve)
 
   int dim = pmesh->Dimension();
 
-  // Define the finite element spaces for displacement field
-  mfem::H1_FECollection fe_coll(1, dim);
-  mfem::ParFiniteElementSpace fe_space(pmesh, &fe_coll, dim, mfem::Ordering::byVDIM);
-
-  // Define a grid function for the global reference configuration, the beginning
-  // step configuration, the global deformation, the current configuration/solution
-  // guess, and the incremental nodal displacements
-  mfem::ParGridFunction x_inc(&fe_space);
-
-  mfem::VectorFunctionCoefficient defo_coef(dim, InitialDeformation);
-  x_inc.ProjectCoefficient(defo_coef);
-  x_inc.SetTrueVector();
+  // Define the solver object
+  NonlinearSolidSolver solid_solver(1, pmesh);
 
   // define a boundary attribute array and initialize to 0
   mfem::Array<int> ess_bdr;
-  ess_bdr.SetSize(fe_space.GetMesh()->bdr_attributes.Max());
+  ess_bdr.SetSize(pmesh->bdr_attributes.Max());
   ess_bdr = 0;
 
   // boundary attribute 1 (index 0) is fixed (Dirichlet)
   ess_bdr[0] = 1;
 
-  mfem::Array<int> trac_bdr;
-  trac_bdr.SetSize(fe_space.GetMesh()->bdr_attributes.Max());
+  // define the displacement vector
+  mfem::Vector disp(dim);
+  disp = 0.0;
+  mfem::VectorConstantCoefficient disp_coef(disp);
 
+
+  mfem::Array<int> trac_bdr;
+  trac_bdr.SetSize(pmesh->bdr_attributes.Max());
   trac_bdr = 0;
   trac_bdr[1] = 1;
 
   // define the traction vector
   mfem::Vector traction(dim);
   traction = 0.0;
   traction(1) = 1.0e-3;
-
   mfem::VectorConstantCoefficient traction_coef(traction);
 
-  // construct the nonlinear mechanics operator
-  QuasistaticSolver oper(fe_space, ess_bdr, trac_bdr,
-                         0.25, 10.0, traction_coef,
-                         1.0e-3, 1.0e-6,
-                         5000, false, false);
+  // Pass the BC information to the solver object
+  solid_solver.SetDisplacementBCs(ess_bdr, &disp_coef);
+  solid_solver.SetTractionBCs(trac_bdr, &traction_coef);
+
+  // Set the material parameters
+  solid_solver.SetHyperelasticMaterialParameters(0.25, 10.0);
+
+  // Set the linear solver params
+  LinearSolverParameters params;
+  params.rel_tol = 1.0e-6;
```

Comment:
Should these magic numbers be in one place? I see a lot of them repeating and my inner CS person cringes.

static constexpr double whatever_it_should_be_called = 1.0e-6;

Reply:
This is more of a general question not to hold up the PR.

Reply:
Unfortunately, these parameters will probably be different for every solve (I typically set them in all of my ALE3D decks). There should be reasonable defaults though. I think I put those in the struct definition?

Reply:
I'll add an issue about this.

Reply:
I will take your word on that.